### PR TITLE
chore: remove `pop_with_default()` function

### PR DESCRIPTION
**Pull request by JohnVillalovos**
In current Pythons we can use `dict.pop(key, default)`

Also converted some `**kwargs` to two named arguments with defaults.

### CODE REVIEW

**Comment by axoroll7** [comment]
Which current Pythons ?

**Comment by JohnVillalovos** [comment]
> Which current Pythons ? Edit : Works with 3.6

All the ones that are supported by the current version of `imapclient`

**Comment by axoroll7** [comment]
> > Which current Pythons ? Edit : Works with 3.6
> 
> All the ones that are supported by the current version of `imapclient`

Ok thanks, I should have known

**Comment by JohnVillalovos** [comment]
As a bit of trivia. It was added in Python 2.3.

https://docs.python.org/2.7/library/stdtypes.html#mapping-types-dict

File: imapclient/imapclient.py [hunk 1]
```diff
@@ -1721,11 +1721,9 @@ def _send_literal(self, tag, item):
         logger.debug("   (literal) > %s", debug_trunc(item, 256))
         self._imap.send(item)
 
-    def _command_and_check(self, command, *args, **kwargs):
-        unpack = pop_with_default(kwargs, "unpack", False)
-        uid = pop_with_default(kwargs, "uid", False)
-        assert not kwargs, "unexpected keyword args: " + ", ".join(kwargs)
```
Comment (mjs) [inline imapclient/imapclient.py:1727]: I think at some point kwargs was passed down to other calls but not any more so what you've done makes much more sense.

**Review by mjs — APPROVED** [review]
A nice little clean up. Thanks.

**Pull request merged**